src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -16,8 +16,96 @@ extern "C" {
 
 }
 
+#include <qurt.h>
+
 #include "../HalideRuntime.h"
 
+const int stack_alignment = 128;
+const int stack_size = 1024 * 1024;
+
+typedef int (*pipeline_argv_t)(void **);
+
+// We can't control the stack size on the thread which receives our
+// FastRPC calls. To work around this, we make our own thread, and
+// forward the calls to that thread.
+class PipelineContext {
+    void *stack;
+    qurt_thread_t thread;
+    qurt_cond_t wakeup_thread;
+    qurt_cond_t wakeup_caller;
+    qurt_mutex_t work_mutex;
+    qurt_mutex_t wakeup_master_mutex;
+
+    // Shared state
+    pipeline_argv_t function;
+    void **args;
+    int result;
+
+    void thread_main() {
+        while (true) {
```

Comment:
Per conversation, this code does not pattern match to how condition variables should be used. LEt me pull this down and do a quick rewrite as a suggestion.